cmd/thanos/compact.go

```diff
@@ -284,9 +284,35 @@ func runCompact(
 			if err == nil {
 				return nil
 			}
+
+			if multiErr, ok := err.(tsdb.MultiError); ok {
+				retriable := true
+
+				for _, err := range multiErr {
+					if compact.IsHaltError(err) {
+						if haltOnError {
+							level.Error(logger).Log("msg", "critical error detected; halting", "err", err)
+							halted.Set(1)
+							select {}
```

Comment:
I'm not sure if this is the best approach? Do we really want to leave pod hanging on the error?

Reply:
@povilasv appreciate the review - this is duplicated code that I've removed now 👍